DigitalLearningSolutions.Web.Tests/Controllers/Login/LoginControllerTests.cs

```diff
@@ -123,6 +123,27 @@ public async Task Successful_sign_in_with_local_return_url_should_redirect_to_re
             result.Should().BeRedirectResult().WithUrl(returnUrl);
         }
 
+        [Test]
+        public async Task Successful_sign_in_with_query_string_should_redirect_with_correct_query()
+        {
+            // Given
+            GivenSignInIsSuccessful();
+
+            var returnUrl = "/some/other/page";
+            var queryString = "?query=true&other=false";
+            var urlHelper = controller.Url;
+            A.CallTo(() => urlHelper.IsLocalUrl(returnUrl)).Returns(true);
+
+            // When
+            var loginViewModel = LoginTestHelper.GetDefaultLoginViewModel();
+            loginViewModel.ReturnUrl = returnUrl;
+            loginViewModel.QueryString = queryString;

```

Comment:
I would put this in `Given` because it's not something you're testing - it's just setting up data.